eventsourcing/src/main/java/org/axonframework/eventsourcing/AnnotationBasedEventStateApplier.java

```diff
@@ -0,0 +1,136 @@
+     */
+    public AnnotationBasedEventStateApplier(@Nonnull Class<M> modelType) {
+        this(modelType, ClasspathParameterResolverFactory.forClass(modelType),
+             new ClassBasedMessageTypeResolver());
+    }
+
+    /**
+     * Initialize a new {@link AnnotationBasedEventStateApplier} for the given model type using the provided
+     * {@code messageTypeResolver}.
+     *
+     * @param modelType           The type of model this instance will handle state changes for.
+     * @param messageTypeResolver The {@link MessageTypeResolver} resolving the
+     *                            {@link org.axonframework.messaging.MessageType types} for
+     *                            {@link org.axonframework.eventhandling.EventMessage EventMessages}
+     */
+    public AnnotationBasedEventStateApplier(@Nonnull Class<M> modelType,
+                                            @Nonnull MessageTypeResolver messageTypeResolver) {
+        this(modelType, ClasspathParameterResolverFactory.forClass(modelType), messageTypeResolver);
+    }
+
+    /**
+     * Initialize a new {@link AnnotationBasedEventStateApplier}.
+     *
+     * @param modelType           The type of model this instance will handle state changes for.
+     * @param parameterResolverFactory the strategy for resolving handler method parameter values
+     * @param messageTypeResolver      The {@link MessageTypeResolver} resolving the
+     *                                 {@link org.axonframework.messaging.MessageType types} for
+     *                                 {@link org.axonframework.eventhandling.EventMessage EventMessages}
+     */
+    private AnnotationBasedEventStateApplier(
+            @Nonnull Class<M> modelType,
+            @Nonnull ParameterResolverFactory parameterResolverFactory,
+            @Nonnull MessageTypeResolver messageTypeResolver) {
+        this(parameterResolverFactory,
+             ClasspathHandlerDefinition.forClass(modelType),
+             messageTypeResolver);
+    }
+
+    /**
+     * Initialize a new {@link AnnotationBasedEventStateApplier}.
+     *
+     * @param parameterResolverFactory the strategy for resolving handler method parameter values
+     * @param handlerDefinition        the handler definition used to create concrete handlers
+     * @param messageTypeResolver      The {@link MessageTypeResolver} resolving the
+     *                                 {@link org.axonframework.messaging.MessageType types} for
+     *                                 {@link org.axonframework.eventhandling.EventMessage EventMessages}
+     */
+    public AnnotationBasedEventStateApplier(@Nonnull ParameterResolverFactory parameterResolverFactory,
+                                            @Nonnull HandlerDefinition handlerDefinition,
+                                            @Nonnull MessageTypeResolver messageTypeResolver) {
+        requireNonNull(parameterResolverFactory, "ParameterResolverFactory may not be null");
```

Comment:
Nit: `requireNonNull` returns the value that the null check was performed on. Hence, you can do the following:
```java
this.parameterResolverFactory = requireNonNull(parameterResolverFactory, "ParameterResolverFactory may not be null");
```